src/java.desktop/share/classes/javax/imageio/spi/IIORegistry.java

```diff
@@ -110,9 +110,7 @@ private IIORegistry() {
      * the Image I/O API.  This instance should be used for all
      * registry functions.
      *
-     * <p> Each {@code ThreadGroup} will receive its own
-     * instance; this allows different {@code Applet}s in the
-     * same browser (for example) to each have their own registry.
+     * <p> Each {@code ThreadGroup} will receive its own instance.
```

Comment:
Note that this new specification is not tied to AppContext. Do we still want to support ThreadGroup sandboxing even if AppContext will be removed? Can we drop it now, since current spec clearly tied to applets and AppContext usage?

Reply:
> I would like to confirm that we treat ThreadGroup based sandboxing as equivalent to AppContext based sandboxing. There are a few places where we still use ThreadGroup instead of AppContext, and I believe we can drop that usage at the same time AppContext is removed, even if the specification includes statements like "Each {@code ThreadGroup} will receive its own instance." - correct?

I *think* that is true. The word/name AppContext probably should never be in API doc and saying  ThreadGroup is a decent substitute, but wherever we did either of these for the sake of applet partitioning they can probably all go away. I hope spec. appearances are rare.
Eliminating AppContext is going to be a good size piece by piece project even without that.